src/ucp/rndv/rndv_mtype.inl

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2021, NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCP_RNDV_MTYPE_INL_
+#define UCP_RNDV_MTYPE_INL_
+
+#include "proto_rndv.inl"
+
+#include <ucp/core/ucp_worker.h>
+
+
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_proto_rndv_mtype_init(const ucp_proto_init_params_t *init_params,
+                          ucp_md_map_t *mdesc_md_map_p, size_t *frag_size_p)
+{
+    ucp_worker_h worker = init_params->worker;
+    ucs_status_t status;
+
+    if ((init_params->select_param->dt_class != UCP_DATATYPE_CONTIG) ||
+        (worker->mem_type_ep[init_params->select_param->mem_type] == NULL)) {
+        return UCS_ERR_UNSUPPORTED;
+    }
+
+    if (mdesc_md_map_p != NULL) {
```

Comment:
is it used somewhere with `NULL` passed as a second parameter?
can this check be removed?